### PR TITLE
[DE249] Destroying all snapshot of rebuild clone dataset once rebuild completes

### PR DESCRIPTION
**Pull request by mynktl**
Changes:
1. [DE249] Destroying all snapshot of rebuild clone dataset once rebuild completes

2. Changes in `zfs stats` output.
     ######                   ...
     ######                   isIOAckSenderCreated: 1
     ######                   isIOReceiverCreated: 1
     ######                   ...
     #####             Earlier it was :
     ######                   ...
     ######                   isIOAckSenderCreated: -1
     ######                   isIOReceiverCreated: -1
     ######                   ...

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

### CODE REVIEW

File: lib/libzpool/uzfs_rebuilding.c [hunk 1]
```diff
@@ -393,3 +393,44 @@ uzfs_zvol_get_or_create_internal_clone(zvol_state_t *zv,
...
+ * on a dataset
+ */
+int
+uzfs_destroy_internal_all_snap(zvol_state_t *zv)
+{
+	int ret;
+	char snapname[MAXNAMELEN];
+	objset_t *os = zv->zv_objset;
+	uint64_t obj = 0, cookie = 0;
+
```
Comment (pawanpraka1) [inline lib/libzpool/uzfs_rebuilding.c:408]: put a null check for zv.

Comment (mynktl) [reply lib/libzpool/uzfs_rebuilding.c:408]: ok

**Review by pawanpraka1 — APPROVED** [review]
looks good.

File: lib/libzpool/uzfs_rebuilding.c [hunk 2]
```diff
@@ -393,3 +393,49 @@ uzfs_zvol_get_or_create_internal_clone(zvol_state_t *zv,
...
+		    snapname, &obj, &cookie, NULL);
+		dsl_pool_config_exit(spa_get_dsl(zv->zv_spa), FTAG);
+
+		if (ret) {
+			if (ret == ENOENT)
+				ret = 0;
+			break;
+		}
+
+		if (!(strcmp(snapname, REBUILD_SNAPSHOT_SNAPNAME) == 0) &&
```
Comment (vishnuitta) [inline lib/libzpool/uzfs_rebuilding.c:426]: can we use internal_snapshot() here?

File: lib/libzpool/uzfs_rebuilding.c [hunk 3]
```diff
@@ -393,3 +393,49 @@ uzfs_zvol_get_or_create_internal_clone(zvol_state_t *zv,
 	strfree(snapname);
 	return (ret);
 }
+
+/*
+ * To destroy all internal created snapshot
+ * on a dataset
+ */
+int
+uzfs_destroy_internal_all_snap(zvol_state_t *zv)
```
Comment (vishnuitta) [inline lib/libzpool/uzfs_rebuilding.c:402]: name change to 'uzfs_destroy_all_internal_snapshots'?

**Review by vishnuitta — COMMENTED** [review]
changes are good.. just minor comments

**Comment by vishnuitta** [comment]
This need to be cherry-picked as well @mynktl

File: lib/libzpool/uzfs_rebuilding.c (hunk 2 above)
Comment (mynktl) [reply lib/libzpool/uzfs_rebuilding.c:426]: `internal_snapshot` is from libzrepl. So, we cannot use it here unless we link libzpool with libzrepl.

File: lib/libzpool/uzfs_rebuilding.c (hunk 3 above)
Comment (mynktl) [reply lib/libzpool/uzfs_rebuilding.c:402]: ok

**Pull request merged**